### DISCUSSION TITLE
Unable to compile rabbitmq-server as unable to build lager dependencies

### DISCUSSION THREAD

**Discussion by rohan-97** [rabbitmq/rabbitmq-server#2423] [Other] [answered]
I am trying to compile and build rabbitmq-server on Debian 9 machine.

I installed dependencies as mentioned [here](https://www.rabbitmq.com/build-server.html)

I cloned the repository and ran make command and got following error. 

```
root@geek:~/source/rabbitmq-server# git status
On branch master
Your branch is up-to-date with 'origin/master'.
nothing to commit, working tree clean
root@geek:~/source/rabbitmq-server# make
make[1]: Entering directory '/home/rohan/source/rabbitmq-server/apps/rabbitmq_prelaunch'
make[2]: Entering directory '/home/rohan/source/rabbitmq-server/deps/rabbit_common'
make[3]: Entering directory '/home/rohan/source/rabbitmq-server/deps/rabbitmq_codegen'
make[3]: Leaving directory '/home/rohan/source/rabbitmq-server/deps/rabbitmq_codegen'
Error: No Makefile to build dependency /home/rohan/source/rabbitmq-server/deps/lager.
erlang.mk:4512: recipe for target 'deps' failed
make[2]: *** [deps] Error 2
make[2]: Leaving directory '/home/rohan/source/rabbitmq-server/deps/rabbit_common'
../../erlang.mk:4512: recipe for target 'deps' failed
make[1]: *** [deps] Error 2
make[1]: Leaving directory '/home/rohan/source/rabbitmq-server/apps/rabbitmq_prelaunch'
erlang.mk:4464: recipe for target 'apps' failed
make: *** [apps] Error 2
root@geek:~/source/rabbitmq-server#
```

I might be following wrong process.
so please suggest the right approach

Thanks

**Comment by lukebakken** [comment, marked as answer]
Here is a transcript from my Arch Linux workstation:

[rabbitmq-server-compile.txt](https://github.com/rabbitmq/rabbitmq-server/files/5022165/rabbitmq-server-compile.txt)

You should begin by ensuring that Erlang and Elixir are installed and in your `PATH`.

**Reply by rohan-97** [reply]
Thanks @lukebakken ,

the dependencies in my debian machine were not correctly configured,
I created a plain debian 9 docker image and again installed all the dependencies and ran make command
and it worked.
Thanks a lot

**Comment by gerhard** [comment]
@rohan-97 this walks you through how to do it from scratch, explains why we start with the https://github.com/rabbitmq/rabbitmq-public-umbrella, how to switch between releases and even how to build & use a Docker image: https://github.com/rabbitmq/tgir/tree/S01E03/s01/e03

**Reply by rohan-97** [reply]
That's cool.
the entire process is explained very nicely.
Thanks @gotthardp
